ERCS/erc-8003.md

```diff
@@ -0,0 +1,147 @@
+---
+eip: 8003
+title: ERC-20 Pre-initialization (Sentinel Storage)
+description: Optional extension to pre-allocate a balance slot via a sentinel bytes32 value so the first SSTORE is prepaid, reducing later transfer gas.
+author: German Maria Abal Bazzano (@ariutokintumi) <g@evvm.org>
+discussions-to: https://ethereum-magicians.org/t/erc-8003-erc-20-pre-initialization-extension-sentinel-storage/24993
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-07-17
+requires: 20
+---
+
+
+## Abstract
+
+This [ERC-20](../EIPS/eip-20.md) extension introduces an **optional pre-initialization function** for ERC-20 tokens, leveraging a sentinel (magic) value and a `bytes32` storage mapping for balances. This mechanism allows users to pay the high SSTORE gas cost in advance (when gas is cheap), so that the later first real token transfer to that address costs far less gas during a potential upcoming network congestion.
```

Comment:
```suggestion
This [ERC-20](./eip-20.md) extension introduces an **optional pre-initialization function** for ERC-20 tokens, leveraging a sentinel (magic) value and a `bytes32` storage mapping for balances. This mechanism allows users to pay the high SSTORE gas cost in advance (when gas is cheap), so that the later first real token transfer to that address costs far less gas during a potential upcoming network congestion.
```